src/librustc_errors/lib.rs

```diff
@@ -907,9 +915,8 @@ impl HandlerInner {
         self.emit_diagnostic(&Diagnostic::new(level, msg));
     }
 
-    fn bug(&mut self, msg: &str) -> ! {
+    fn bug(&mut self, msg: &str) {
         self.emit_diagnostic(&Diagnostic::new(Bug, msg));
-        panic!(ExplicitBug);
```

Comment:
The change here doesn't seem to account for `.emit_error` above which is called by `fn err` and `fn fatal`.